src/seq/nucleotide.jl

```diff
@@ -103,6 +103,24 @@ function isGC(nt::DNANucleotide)
 end
 
 """
+    isPurine(nt::Nucleotide)
+
+Test if nucleotide is surely a Purine.
+"""
+function isPurine(nt::Nucleotide)
+    return (compatbits(nt) | 0b0101) == 0b0101
```

Comment:
Also note that `DNA_R`, an ambiguous DNA nucleotide, is surely a purine because R = A or G.